client/src/components/StarsComponent.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { StyleSheet, TouchableOpacity, View } from 'react-native';
+import Icon from 'react-native-vector-icons/Ionicons';
+
+import { Colors } from '../assets/Colors';
+
+type RatingComponentPropsType = {
+    rate: number,
+    setRate?: (rate: number) => void,
+    iconSize?: number
+}
+
+enum StarsEnum {
+    half = 'star-half-sharp',
+    outline = 'star-outline',
+    sharp = 'star-sharp'
+}
+
+const stars = [1, 2, 3, 4, 5];
+
+export const StarsComponent = (props: RatingComponentPropsType) => {
+    const { rate, setRate, iconSize = 25 } = props;
+    const fullStars = Math.trunc(rate);
+    const halfStar = Math.ceil(rate);
+
+    const getStarName = (i: number) => {
+        if (i <= fullStars) {
+            return StarsEnum.sharp;
+        } else if (i === halfStar) {
+            return StarsEnum.half;
+        } else {
+            return StarsEnum.outline;
+        }
+    };
```

Comment:
you can just pass fullStars as second argument and move this function out of component to make it smaller